### PR TITLE
stm32f4: adc: Correct ADC documentation

### PR DESCRIPTION
**Pull request by 0xdec**
Fix the demo code, add info about different connections for STM32F42x/STM32F43x, and remove errant param annotations.

### CODE REVIEW

File: lib/stm32/f4/adc.c [hunk 1]
```diff
@@ -47,17 +47,17 @@ independent.
...
 @code
 gpio_mode_setup(GPIOA, GPIO_MODE_ANALOG, GPIO_PUPD_NONE, GPIO1);
-rcc_peripheral_enable_clock(&RCC_APB2ENR, RCC_APB2ENR_ADC1EN);
-adc_set_clk_prescale(RCC_CFGR_ADCPRE_BY2);
+rcc_periph_clock_enable(RCC_ADC1);
+adc_set_clk_prescale(ADC_CCR_ADCPRE_BY2);
 adc_disable_scan_mode(ADC1);
 adc_set_single_conversion_mode(ADC1);
-adc_set_sample_time(ADC1, ADC_CHANNEL0, ADC_SMPR1_SMP_1DOT5CYC);
+adc_set_sample_time(ADC1, ADC_CHANNEL0, ADC_SMPR_SMP_3CYC);
```
Comment (karlp) [inline lib/stm32/f4/adc.c]: probably better here would be explaining how to choose the value, or simply putting in "xxxxx" :)

File: lib/stm32/f4/adc.c [hunk 2]
```diff
@@ -406,13 +406,13 @@ bool adc_awd(uint32_t adc)
 /*---------------------------------------------------------------------------*/
 /** @brief ADC Enable The Temperature Sensor
 
-This enables both the sensor and the reference voltage measurements on channels
-16 and 17. These are only available on ADC1 channel 16 and 17 respectively.
-
-@param[in] adc Unsigned int32. ADC block register address base @ref adc_reg_base
```
Comment (karlp) [inline lib/stm32/f4/adc.c]: Thanks for this,  I saw it in the context of your earlier diff and was going to strip it out when I got to it.

**Comment by karlp** [comment]
looks reasonable enough, though tempting to remove that intro section altogether and simply refer to an example project.

File: lib/stm32/f4/adc.c (hunk 1 above)
Comment (0xdec) [inline lib/stm32/f4/adc.c]: Okay, I'll just put xxxx

**Comment by 0xdec** [comment]
Yeah I figured I would just leave it in because I think it's still useful to be able to see example code without an internet connection.

File: lib/stm32/f4/adc.c (hunk 1 above)
Comment (0xdec) [inline lib/stm32/f4/adc.c]: Added a short explanation

**Comment by karlp** [comment]
pulled into my "next" branch, pure cleanup, nothing really to hold this up

**Comment by karlp** [comment]
landed via c7d46c4f thanks

**Pull request closed without merging**